### PR TITLE
Fix/binary linear model wrapper fix

### PR DESCRIPTION
**Pull request by AlessandroPolidori**
## Summary

Problem fixed: when num_classes=2, logistic regression returns one-dimensional outputs and this is problematic for the pytorch linear models wrapper. Solved giving the original output to a Sigmoid and building 2-dimensional outputs.

## Type of Change

Please select the one relevant option below:

- Bug fix (non-breaking change that solves an issue)

## Checklist

Please confirm that the following tasks have been completed:

- [x] I have tested my changes locally and they work as expected. (Please describe the tests you performed.)
- [x] I have added unit tests for my changes, or updated existing tests if necessary.
- [x] I have updated the documentation, if applicable.
- [x] I have installed pre-commit and run locally for my code changes.

### CODE REVIEW

File: quadra/utils/vit_explainability.py [hunk 1]
```diff
@@ -303,4 +303,10 @@ def __init__(
         self.classifier.bias.data = torch.from_numpy(linear_classifier.intercept_).float()
 
     def forward(self, x):
+        if self.num_classes == 2:
+            sig = torch.nn.Sigmoid()
+            class_zero_probabilities = 1 - sig(self.classifier(self.backbone(x)))
+            class_one_probabilities = sig(self.classifier(self.backbone(x)))
```
Comment (lorenzomammana) [inline quadra/utils/vit_explainability.py]: Two forward passages to reach the same result of a single one >.>

File: quadra/utils/vit_explainability.py [hunk 2]
```diff
@@ -303,4 +303,10 @@ def __init__(
...
+            two_class_probs = torch.cat((class_zero_probabilities, class_one_probabilities), dim=1)
+            return torch.nn.Softmax(dim=1)(two_class_probs)
```
Comment (lorenzomammana) [inline quadra/utils/vit_explainability.py:312]: Why we are applying Softmax again? Sigmoid already maps in 0-1

**Review by lorenzomammana — CHANGES_REQUESTED** [review]

**Review by lorenzomammana — APPROVED** [review]
Approved
